packages/wrangler/src/deprecated/index.ts

```diff
@@ -47,19 +41,22 @@ export function generateOptions(yargs: Argv<CommonYargsOptions>) {
 			deprecated: true,
 		});
 }
+type GenerateArgs = YargsOptionsToInterface<typeof generateOptions>;
 
-export async function generateHandler({
+//TODO: move this out of deprecated since apparently it's not deprecated anymore
+export async function generateHandler(args: GenerateArgs) {
 	// somehow, `init` marks name as required but then also runs fine
 	// with the name omitted, and then substitutes it at runtime with ""
-	name = "",
-	template,
-	type,
-	site,
-	...args
-}: ArgumentsCamelCase<GenerateArgs>) {
+	let { name = "", template } = args;
+	const { type, site } = args;
 	// delegate to `wrangler init` if no template is specified
 	if (template === undefined) {
-		return initHandler({ name, ...args });
+		return initHandler({
+			...args,
```

Comment:
Personally I would avoid the shortcut here and just copy out the args by hand.
It might feel a bit brittle since if a new property is added to the initHandler params this may miss it; but the other side of the coin is that this way a new parameter added to generateHandler will quietly fall through.
There is no idea approach I think. But being more explicit about what we "want" feels safer.

Reply:
Yeah good call